### PR TITLE
index/blog: list authors

### PR DESCRIPTION
**Pull request by leandrolanzieri**
## Contribution description
This displays the list of authors (when available) both in the blog entry header and in the index boxes.

Currently an author can define:
- name
- email
- githubid
- affiliation

But this can easily be extended in the future.

### CODE REVIEW

File: _posts/2020-12-31-test.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ title:  "Happy Bobo"
 description: "Lorem markdownum e Daphnes convocat perque medios, pro nigra totiens. Maledicere tenebas. Manus subit moderantum sunt iamdudum qui exsilit dictis forte numen vulgus facilem. Acta honore eandem"
 image: bobo.jpg
 date: 20/10/2020
+authors:
+    - name: "Bobo Mono"
+      email: "bobo@riot-os.org"
+      githubid: "riot-ci"
```
Comment (miri64) [inline _posts/2020-12-31-test.md:10]: Should we maybe also add a link to the forum?

File: _layouts/blog.html [hunk 2]
```diff
@@ -12,6 +12,19 @@
       <div class="row pt-4 d-flex align-items-center text-center h-100">
         <div class="col-12 text-white">
           <h1 class="display-3 w-100"> {{ page.title }} </h1>
+          <h4>
+            {% for author in page.authors %}
+              {% if forloop.first %} by {% endif %}
+              {% if author.email %}<a href="mailto:{{ author.email }}">{% endif %}
+              {{ author.name }}
```
Comment (miri64) [inline _layouts/blog.html:19]: What if `author.name` is not supplied?

**Review by miri64 — COMMENTED** [review]
Does what it should do and looks really nice :-). However, there is a question.

File: _layouts/blog.html (hunk 2 above)
Comment (leandrolanzieri) [reply _layouts/blog.html:19]: If there is no `authors` this should work. If not providing a name is actually a concern I can simply add a default. Suggestions?

File: _posts/2020-12-31-test.md (hunk 1 above)
Comment (waehlisch) [reply _posts/2020-12-31-test.md:10]: not in favor of this as i don't see this such links as common.  you can refer in the text if needed.

it is rather more likley that you link from the forum to thie blog article.

File: _layouts/blog.html (hunk 2 above)
Comment (miri64) [reply _layouts/blog.html:19]: No I meant actually some kind of error case where the author e.g. only provides the other keys or misspells the name key.

File: _posts/2020-12-31-test.md (hunk 1 above)
Comment (miri64) [reply _posts/2020-12-31-test.md:10]: But what if the reader prefers to contact the author via discourse DM? Or the author prefers a discourse DM vs publishing their email address?

Comment (miri64) [reply _posts/2020-12-31-test.md:10]: At any rate: all of these should be kept optional of course, I just want to give authors the ability to be contacted the way they prefer in a manner that integrates nicely into our blog.

File: _layouts/blog.html (hunk 2 above)
Comment (leandrolanzieri) [reply _layouts/blog.html:19]: Jekyll would not fail, it will be empty. Any preference on a placeholder?

File: _posts/2020-12-31-test.md (hunk 1 above)
Comment (miri64) [reply _posts/2020-12-31-test.md:10]: We also could add more options, like they do here: https://qfimg.github.io/author-box-jekyll/

File: _layouts/blog.html (hunk 2 above)
Comment (miri64) [reply _layouts/blog.html:19]: "Anonymous"?

Comment (leandrolanzieri) [reply _layouts/blog.html]: Added.

File: _posts/2020-12-31-test.md (hunk 1 above)
Comment (leandrolanzieri) [reply _posts/2020-12-31-test.md:10]: Extended to support the following:
- web
- twitter
- stack overflow
- reddit
- linkedin
- mastodon
- facebook

Comment (miri64) [reply _posts/2020-12-31-test.md:10]: Still no discourse :disappointed:, but okay.

**Review by miri64 — APPROVED** [review]
Looking good. Please squash!

**Comment by leandrolanzieri** [comment]
Squashed!

File: _layouts/blog.html [hunk 3]
```diff
@@ -12,6 +12,50 @@
...
+              {% if author.githubid %}
+                <a href="https://github.com/{{ author.githubid }}" target="_blank"><i class="ri-github-fill align-text-bottom"></i></a>
+              {% endif %}
+
+              {% if author.twitterid %}
+                <a href="https://twitter.com/{{ author.twitterid }}" target="_blank"><i class="ri-twitter-fill align-text-bottom"></i></a>
+              {% endif %}
+
+              {% if author.stackoverflowid %}
+                <a href="https://stackoverflow.com/u/{ author.stackoverflowid }}" target="_blank"><i class="ri-stack-overflow-fill align-text-bottom"></i></a>
```
Comment (miri64) [inline _layouts/blog.html:36]: Oops, only just noticed that there is a `{` missing.

File: _layouts/blog.html [hunk 4]
```diff
@@ -12,6 +12,50 @@
...
+              {% endif %}
+
+              {% if author.redditid %}
+                <a href="https://reddit.com/user/{{ author.redditid }}" target="_blank"><i class="ri-reddit-fill align-text-bottom"></i></a>
+              {% endif %}
+
+              {% if author.linkedinid %}
+                <a href="https://linkedin.com/in/{ author.linkedinid }}" target="_blank"><i class="ri-linkedin-box-fill align-text-bottom"></i></a>
```
Comment (miri64) [inline _layouts/blog.html:44]: Here too

File: _layouts/blog.html (hunk 3 above)
Comment (leandrolanzieri) [reply _layouts/blog.html:36]: Will fix

Comment (leandrolanzieri) [reply _layouts/blog.html:36]: See #32

**Pull request merged**